### PR TITLE
Tweaks to "getting started" documentation

### PR DESCRIPTION
**Pull request by rswarbrick**
These are mostly changes to phrasing to make things easier to understand.

The only substantial change is to drop the stuff about overly old GCC on the (now obsolete) Ubuntu 18.

### CODE REVIEW

File: doc/guides/getting_started/src/README.md [hunk 1]
```diff
@@ -42,16 +45,16 @@ If you do have Linux, you are still welcome to try the Docker container.
...
 You will need at least **7GiB of available RAM** in order to build the Verilator simulation.
-If you have an FPGA and download the bitstream from our cloud bucket rather than building it locally (the default setup) then this constraint does not apply.
+If you are building another form of simulation, this constraint does not apply.
 
 If you are specifying a new machine to run top-level simulations of the whole of OpenTitan using Verilator, it is recommended that you
-have a minimum of **32GiB of physical RAM** and at least **512GiB of SSD/HDD storage** for the build tools, repository and Ubuntu installation.
+have a minimum of **32GiB of physical RAM** and at least **512GiB of disk storage** for the build tools, repository and Ubuntu installation.
 
-## Step 2: Install Package Manager Dependencies
+## Step 2: Install dependencies using the package manager
```
Comment (mundaym) [inline doc/guides/getting_started/src/README.md:57]: Low priority but do you think you could go through all the headings and change them to use a consistent capitalization scheme?

File: doc/guides/getting_started/src/README.md [hunk 2]
```diff
@@ -221,22 +194,22 @@ After installation you need to add `/tools/verible/$VERIBLE_VERSION/bin` to your
 
 Note that we currently use version {{#tool-version verible }}, but it is expected that this version is going to be updated frequently, since the tool is under active development.
 
-### Step 6b: Install OpenOCD (optional)
+### step 7b: Install OpenOCD (optional)
```
Comment (mundaym) [inline doc/guides/getting_started/src/README.md:197]: Did you mean to make these lower case? Is it so that they are clearer as sub steps?

**Review by mundaym — APPROVED** [review]

File: doc/guides/getting_started/src/README.md (hunk 2 above)
Comment (rswarbrick) [reply doc/guides/getting_started/src/README.md]: Oops! Fixed now, thanks.

File: doc/guides/getting_started/src/README.md (hunk 1 above)
Comment (rswarbrick) [reply doc/guides/getting_started/src/README.md:57]: Hmm, good point. I've done a follow-up with that change here: https://github.com/lowRISC/opentitan/pull/17969.

**Pull request merged**